File: app/src/main/java/com/unis/gameplatfrom/constant/LoginConstant.java
package com.unis.gameplatfrom.constant;

import com.unis.gameplatfrom.entity.LoginEntity;
import com.unis.gameplatfrom.model.BaseModel;
import com.unis.gameplatfrom.ui.view.BaseView;

import io.reactivex.Observable;

public interface LoginConstant {

    interface Model extends BaseModel {
        Observable<LoginEntity> login(String account, String password,
                                                        String netaddress, int type, String key);
    }

    interface View extends BaseView {
        @Override
        void showLoading();

        @Override
        void hideLoading();

        @Override
        void onError(Throwable throwable);

        void onSuccess(LoginEntity result);
    }

    interface Presenter {
        /**
         * 登陆
         *
         * @param account
         * @param password
         */
        void login(String account, String password,
                   String netaddress, int type, String key);
    }




}

File: kotlin/src/main/java/com/unis/kotlin/constant/LoginConstant.kt
package com.unis.kotlin.constant

import com.unis.kotlin.entity.LoginEntity
import com.unis.kotlin.model.BaseModel
import com.unis.kotlin.ui.view.BaseView
import io.reactivex.Observable

interface LoginConstant{

    interface Model : BaseModel{

        abstract fun login(account: String, password: String,
                           netaddress: String, type: Int, key: String): Observable<LoginEntity>

    }

    interface View : BaseView{

        override fun hideLoading() {

        }

        override fun showLoading() {

        }

        override fun onError(throwable: Throwable) {

        }

        fun onSuccess(result: LoginEntity){}

    }

    interface Presenter{

        /**
         * 登陆
         *
         * @param account
         * @param password
         */
         fun login(account: String, password: String,
                           netaddress: String, type: Int, key: String)

    }





}